catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/QueryOperations.java

```diff
@@ -0,0 +1,622 @@
+
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.Constants;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.federation.FederationStrategy;
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.Request;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.ProcessingDetailsImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.operation.impl.SourceResponseImpl;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.PreQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.ConnectedSource;
+import ddf.catalog.source.FederatedSource;
+import ddf.catalog.source.Source;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.util.impl.DescribableImpl;
+import ddf.catalog.util.impl.Requests;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.common.audit.SecurityLogger;
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValueCollectionPermission;
+
+public class QueryOperations extends DescribableImpl {
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryOperations.class);
+
+    private Supplier<CatalogProvider> catalogSupplier;
+
+    // Inject properties
+    private FrameworkProperties frameworkProperties;
+
+    private SourceOperations sourceOperations;
+
+    private OperationsSecuritySupport opsSecuritySupport;
+
+    private OperationsMetacardSupport opsMetacardSupport;
+
+    public QueryOperations(FrameworkProperties frameworkProperties,
+            SourceOperations sourceOperations, OperationsSecuritySupport opsSecuritySupport,
+            OperationsMetacardSupport opsMetacardSupport) {
+        this.frameworkProperties = frameworkProperties;
+        this.sourceOperations = sourceOperations;
+        this.opsSecuritySupport = opsSecuritySupport;
+        this.opsMetacardSupport = opsMetacardSupport;
+    }
+
+    void setCatalogSupplier(Supplier<CatalogProvider> catalogSupplier) {
+        this.catalogSupplier = catalogSupplier;
+    }
+
+    //
+    // Delegate methods
+    //
+    public QueryResponse query(QueryRequest fedQueryRequest, boolean fanoutEnabled)
```

Comment:
Yeah that would work, I should have explained that at a glance it just seemed like fanoutEnabled was only the concern of this class so it shouldn't be exposed if thats an invalid assumption then it can stay public also.